src/sqlfluff/rules/L003.py

```diff
@@ -416,358 +466,220 @@ def _process_current_line(
         - When passed the *final* segment in the entire parse tree (which may
           not be a newline)
         """
-        this_line_no = max(res.keys())
-        this_line = res.pop(this_line_no)
+        line_summaries = memory.line_summaries
+        this_line_no = max(line_summaries.keys())
+        this_line: _LineSummary = line_summaries.pop(this_line_no)
         self.logger.debug(
-            "Evaluating line #%s. %s",
-            this_line_no,
-            # Don't log the line or indent buffer, it's too noisy.
-            self._strip_buffers(this_line),
+            "Evaluating line #{this_line_no:02}. %s",
+            this_line,
         )
-        trigger_segment = memory["trigger"]
 
-        # Is this line just comments? (Disregard trailing newline if present.)
-        check_comment_line = this_line["line_buffer"]
-        if check_comment_line and all(
-            seg.is_type(
-                "whitespace", "comment", "indent"  # dedent is a subtype of indent
-            )
-            for seg in check_comment_line
-        ):
+        if this_line.is_comment_line:
             # Comment line, deal with it later.
-            memory["comment_lines"].append(this_line_no)
-            self.logger.debug("    Comment Line. #%s", this_line_no)
+            memory.comment_lines.append(this_line_no)
+            self.logger.debug(f"    Comment Line. #{this_line_no:02}")
```

Comment:
You don't want to use f-strings for logging:

The reason is, this will build the log string even if it's not going to log it (i.e. due to log level). The old-style formatting with `%s` delays building the string until it knows it's going to output that string. Definite performance impact.